osm2streets/src/road.rs

```diff
@@ -219,7 +259,8 @@ impl Road {
         // If there's a sidewalk on only one side, adjust the true center of the road.
         // TODO I don't remember the rationale for doing this in the first place. What if there's a
         // shoulder and a sidewalk of different widths? We don't do anything then
-        let mut true_center = self.untrimmed_center_line.clone();
+        //TODO use self.left_edge_offset_of(RoadPosition::Center) or something?
+        let mut true_center = self.reference_line.clone();
```

Comment:
I hope you can figure out the correct semantics here, depending on how `untrimmed_geometry` is used.

Reply:
`untrimmed_road_geometry` is one of these weird bits of legacy code that I don't remember the original reason for. It just looks for cases where a road has a sidewalk on exactly one side, and shifts over the center line a bit. Most of the uses are in transformations just to check length, and those places could use `reference_line.length()` today instead. And once we maintain `center_line` at all times, most/all probably should use that.

The one "real" use of this is in the very beginning of `transform/intersection_geometry.rs`. We initialize a road's `center_line` to this. I _think_ we want to ditch this method entirely and just use the corrected center that's now calculated from placement.

Reply:
I think I understand what is going on with all of these. It looks like the extra logic here is trying to pretend the line is placed at `RoadPosition::Center`, so I have implemented it to behave like that. I have touched all the uses of this fn (to add the `driving_side` param), but mostly left them as is. Ideally we replace the uses with more appropriate lookups as we go.